### PR TITLE
Add support for ghc-9.2 and partial support for CHaP

### PR DESCRIPTION
**Pull request by erikd**

### CODE REVIEW

File: cardano-testnet/src/Testnet/Byron.hs [hunk 1]
```diff
@@ -237,7 +237,7 @@ testnet testnetOptions H.Conf {..} = do
     sprocket <- H.noteShow $ Sprocket tempBaseAbsPath (socketDir </> "node-" <> si)
     _spocketSystemNameFile <- H.noteShow $ IO.sprocketSystemName sprocket
     -- TODO: Better error message need to indicate a sprocket was not created
-    H.byDeadlineM 10 deadline $ H.assertM $ H.doesSprocketExist sprocket
+    H.byDeadlineM 10 deadline "Byron" $ H.assertM $ H.doesSprocketExist sprocket
```
Comment (newhoggy) [inline cardano-testnet/src/Testnet/Byron.hs:240]: This should say "Failed to connect to node socket".

File: cardano-testnet/src/Testnet/Cardano.hs [hunk 2]
```diff
@@ -744,7 +744,7 @@ cardanoTestnet testnetOptions H.Conf {..} = do
   forM_ allNodeNames $ \node -> do
     sprocket <- H.noteShow $ Sprocket tempBaseAbsPath (socketDir </> node)
     _spocketSystemNameFile <- H.noteShow $ IO.sprocketSystemName sprocket
-    H.byDeadlineM 10 deadline $ H.assertM $ H.doesSprocketExist sprocket
+    H.byDeadlineM 10 deadline "Cardano" $ H.assertM $ H.doesSprocketExist sprocket
```
Comment (newhoggy) [inline cardano-testnet/src/Testnet/Cardano.hs:747]: This should say "Failed to connect to node socket".

File: cardano-testnet/src/Testnet/Shelley.hs [hunk 3]
```diff
@@ -417,7 +417,7 @@ shelleyTestnet testnetOptions H.Conf {..} = do
   forM_ allNodes $ \node -> do
     sprocket <- H.noteShow $ Sprocket tempBaseAbsPath (socketDir </> node)
     _spocketSystemNameFile <- H.noteShow $ IO.sprocketSystemName sprocket
-    H.byDeadlineM 10 deadline $ H.assertM $ H.doesSprocketExist sprocket
+    H.byDeadlineM 10 deadline "Shelley" $ H.assertM $ H.doesSprocketExist sprocket
```
Comment (newhoggy) [inline cardano-testnet/src/Testnet/Shelley.hs:420]: This should say "Failed to connect to node socket".

File: cardano-testnet/test/Test/Cli/Babbage/LeadershipSchedule.hs [hunk 4]
```diff
@@ -118,7 +118,7 @@ hprop_leadershipSchedule = H.integration . H.runFinallies . H.workspace "alonzo"
 
     leadershipScheduleDeadline <- H.noteShowM $ DTC.addUTCTime 180 <$> H.noteShowIO DTC.getCurrentTime
 
-    H.byDeadlineM 5 leadershipScheduleDeadline $ do
+    H.byDeadlineM 5 leadershipScheduleDeadline "Line 121" $ do
```
Comment (newhoggy) [inline cardano-testnet/test/Test/Cli/Babbage/LeadershipSchedule.hs:121]: This should say "Failed to query for leadership schedule".

File: cardano-testnet/test/Test/Cli/Babbage/LeadershipSchedule.hs [hunk 5]
```diff
@@ -159,7 +159,7 @@ hprop_leadershipSchedule = H.integration . H.runFinallies . H.workspace "alonzo"
 
     leadershipScheduleDeadline <- H.noteShowM $ DTC.addUTCTime 180 <$> H.noteShowIO DTC.getCurrentTime
 
-    H.byDeadlineM 5 leadershipScheduleDeadline $ do
+    H.byDeadlineM 5 leadershipScheduleDeadline "Line 162" $ do
```
Comment (newhoggy) [inline cardano-testnet/test/Test/Cli/Babbage/LeadershipSchedule.hs:162]: This should say "Failed to query for leadership schedule".

File: cardano-testnet/test/Test/Cli/KesPeriodInfo.hs [hunk 6]
```diff
@@ -186,7 +186,7 @@ hprop_kes_period_info = H.integration . H.runFinallies . H.workspace "chairman"
                , "--testnet-magic", show @Int testnetMagic
                ]
 
-  delegsAndRewards <- H.byDurationM 3 12 $ do
+  delegsAndRewards <- H.byDurationM 3 12 "Line 189" $ do
```
Comment (newhoggy) [inline cardano-testnet/test/Test/Cli/KesPeriodInfo.hs:189]: This should say "Pledge's stake address was not registered"

File: cardano-testnet/test/Test/Cli/KesPeriodInfo.hs [hunk 7]
```diff
@@ -258,7 +258,7 @@ hprop_kes_period_info = H.integration . H.runFinallies . H.workspace "chairman"
 
   H.note_ $ "Check to see if " <> utxoStakingVkey2 <> " was registered..."
 
-  userSAddr <- H.byDurationM 3 12 $ do
+  userSAddr <- H.byDurationM 3 12 "Line 261" $ do
```
Comment (newhoggy) [inline cardano-testnet/test/Test/Cli/KesPeriodInfo.hs:261]: This should say "Failed to query stake address info"

File: cardano-testnet/test/Test/Cli/KesPeriodInfo.hs [hunk 8]
```diff
@@ -332,7 +332,7 @@ hprop_kes_period_info = H.integration . H.runFinallies . H.workspace "chairman"
     , "--testnet-magic", show @Int testnetMagic
     ]
 
-  H.byDurationM 3 12 $ do
+  H.byDurationM 3 12 "Line 335" $ do
```
Comment (newhoggy) [inline cardano-testnet/test/Test/Cli/KesPeriodInfo.hs:335]: This should say "Stake pool was not registered"

**Comment by angerman** [comment]
I think we need to 
```
allow-newer: katip:Win32
```
🤦

**Comment by angerman** [comment]
(rebased onto master)
Will try to get CI green now.

**Comment by angerman** [comment]
## cicero breakage:
- hls (stylish and hls can't agree on the plugin version (1.4 vs 1.5)
- cardano-api fails to link against lib sodium, yet depends on praos vrf.

## Buidlkite runs out of space

**Comment by jbgi** [comment]
rebased again

**Comment by angerman** [comment]
- The failure of the Darwin, 9.2.5 job seems genuine and concerning? https://github.com/input-output-hk/cardano-node/actions/runs/4038755984/jobs/6944927370

The cicero logs are still fairly useless (_way_ too much noise). -- the key seems to be in Loki to set
- `line contains`: `failed`
- `line does not contain`: `dependencies of derivation`

- [ ] `cardano-tracer-test-cardano-tracer-test-x86_64-w64-mingw32> cardano-tracer-test.exe: SetEvent: Invalid handle.`
- [ ] `plutus-tx-plugin-lib-plutus-tx-plugin-x86_64-w64-mingw32>     Neither ‘FamInstEnv’ nor ‘GhcPlugins’ exports ‘ModBreaks’.`
- [ ] `plutus-core-lib-plutus-core-x86_64-unknown-linux-musl>     <command line>: Error loading shared library libgcc_s.so.1: No such file or directory (needed by /nix/store/m91kc99f2qmmdxnb9n5va24mmd5zm8x2-x86_64-unknown-linux-musl-stage-final-gcc-11.3.0-lib/x86_64-unknown-linux-musl/lib/libstdc++.so)`
- [ ] `locli-exe-locli>     Module ‘Cardano.Prelude’ does not export ‘option’`
- [ ] `cardano-client-demo-exe-stake-credential-history>     but were not needed for compilation: - cardano-slotting-0.1.0.2-Hu4biqduZaZLVuG4T3h5se`

**Comment by angerman** [comment]
Sigh, loki and cicero logs are useless now.

**Comment by angerman** [comment]
I believe the remaining failure is primarily:
```
x86_64-darwin	cardano-testnet-test-cardano-testnet-tests>         ExceptionInLinkedThread (ThreadId 404) pokeSockAddr: path is too long
```

**Comment by angerman** [comment]
The failure is remarkable stupid:
```
cardano-testnet-test-cardano-testnet-tests>       stake-snapshot:               OK (172.19s)
cardano-testnet-test-cardano-testnet-tests>           ✓ stake-snapshot passed 1 test.
cardano-testnet-test-cardano-testnet-tests>     kes-period-info:                IGNORED on Disabled
cardano-testnet-test-cardano-testnet-tests>       IGNORED on Disabled
cardano-testnet-test-cardano-testnet-tests>   foldBlocks receives ledger state: FAIL (18.75s)
cardano-testnet-test-cardano-testnet-tests>       ✗ foldBlocks receives ledger state failed at src/Hedgehog/Extras/Test/Base.hs:122:32
cardano-testnet-test-cardano-testnet-tests>         after 1 test.
cardano-testnet-test-cardano-testnet-tests>         shrink path: 1:
cardano-testnet-test-cardano-testnet-tests>         ━━━ Exception (SomeAsyncException) ━━━
cardano-testnet-test-cardano-testnet-tests>         ExceptionInLinkedThread (ThreadId 392) pokeSockAddr: path is too long
cardano-testnet-test-cardano-testnet-tests>         CallStack (from HasCallStack):
cardano-testnet-test-cardano-testnet-tests>           error, called at Network/Socket/Types.hsc:1154:40 in network-3.1.2.7-CpmVedU1bJc7Um4m8fCGkj:Network.Socket.Types
cardano-testnet-test-cardano-testnet-tests>         This failure can be reproduced by running:
cardano-testnet-test-cardano-testnet-tests>         > recheckAt (Seed 15338231145022280194 15780688854180360437) "1:" foldBlocks receives ledger state
cardano-testnet-test-cardano-testnet-tests>     Use '--pattern "$NF ~ /foldBlocks receives ledger state/" --hedgehog-replay "1: Seed 15338231145022280194 15780688854180360437"' to reproduce from the command-line.
cardano-testnet-test-cardano-testnet-tests>     Use -p '/foldBlocks receives ledger state/' to rerun this test only.
cardano-testnet-test-cardano-testnet-tests> 3 out of 6 tests failed (499.07s)
```

**Comment by angerman** [comment]
bors try

**Comment by angerman** [comment]
bors try

**Comment by angerman** [comment]
Bors try

**Review by angerman — APPROVED** [review]
LGTM, let's do this!

**Comment by angerman** [comment]
bors try

**Comment by erikd** [comment]
bors merge

**Comment by angerman** [comment]
bors merge

**Pull request merged**